source/agora/node/Ledger.d

```diff
@@ -115,20 +113,24 @@ public class Ledger
     {
         auto txs = this.pool.take(Block.TxsInBlock);
         assert(txs.length == Block.TxsInBlock);
-        auto block = makeNewBlock(*this.last_block, txs);
+        Block last_block;
+        assert(this.storage.readLastBlock(last_block));
```

Comment:
```d
if (!this.storage.readLastBlock(last_block))
    assert(0);
```

The reason it is written this way is that `assert` are removed in `release` mode.
Which means `readLastBlock` would disappear (!) and the program would not perform correctly.
With `if` + `assert(0)` this does not happen, as `assert(0)` are never removed.